royalties-registry/contracts/RoyaltiesRegistry.sol

```diff
@@ -93,34 +95,55 @@ contract RoyaltiesRegistry is IRoyaltiesProvider, OwnableUpgradeable {
     }
 
     function royaltiesFromContract(address token, uint tokenId) internal view returns (LibPart.Part[] memory) {
+        (LibPart.Part[] memory royalties, bool ok) = royaltiesFromContractNative(token, tokenId);
+        if (!ok) {
+            return royaltiesFromContractSpecial(token, tokenId);
+        }
+        return royalties;
+    }
+
+    function royaltiesFromContractNative(address token, uint tokenId) internal view returns (LibPart.Part[] memory, bool) {
+        bool ok = true;
         if (IERC165Upgradeable(token).supportsInterface(LibRoyaltiesV2._INTERFACE_ID_ROYALTIES)) {
             RoyaltiesV2 v2 = RoyaltiesV2(token);
             try v2.getRaribleV2Royalties(tokenId) returns (LibPart.Part[] memory result) {
-                return result;
+                return (result, ok);
             } catch {}
         } else {
             RoyaltiesV1 v1 = RoyaltiesV1(token);
             address payable[] memory recipients;
             try v1.getFeeRecipients(tokenId) returns (address payable[] memory result) {
                 recipients = result;
             } catch {
-                return new LibPart.Part[](0);
+                return (new LibPart.Part[](0), !ok);
             }
             uint[] memory values;
             try v1.getFeeBps(tokenId) returns (uint[] memory result) {
                 values = result;
             } catch {
-                return new LibPart.Part[](0);
+                return (new LibPart.Part[](0), !ok);
             }
             if (values.length != recipients.length) {
-                return new LibPart.Part[](0);
+                return (new LibPart.Part[](0), ok);
             }
             LibPart.Part[] memory result = new LibPart.Part[](values.length);
             for (uint256 i = 0; i < values.length; i++) {
                 result[i].value = uint96(values[i]);
                 result[i].account = recipients[i];
             }
-            return result;
+            return (result, ok);
+        }
+        return (new LibPart.Part[](0), !ok);
```

Comment:
Using such structure as 
if { return }
else { return }
return
is not transparent to see flow how can we reach the last return. We can reach it in "catch" in "if", thus it's better to place this "return" to "catch" and we will get a transparent structure as
if { return }
else { return }

In "royaltiesFromContractSpecial" is the same proposal

Reply:
Add return in catch, but if there is no return in the end of method, I got warning (no return statement).
Return in the end of method is not deleted

Reply:
delete else looks good